src/tests/unit/tests/DetailsView/components/report-export-component-props-factory.test.ts

```diff
@@ -0,0 +1,165 @@
+
+describe('ReportExportComponentPropsFactory', () => {
+    const theDate = new Date(2019, 2, 12, 9, 0);
+    const theTimestamp = 'test timestamp';
+    const thePageTitle = 'command-bar-test-tab-title';
+    const theDescription = 'test description';
+    const theGeneratorOutput = 'generator output';
+    const thePageUrl = 'test page url';
+
+    let assessmentsProviderMock: IMock<AssessmentsProvider>;
+    let featureFlagStoreData: FeatureFlagStoreData;
+    let actionMessageCreatorMock: IMock<DetailsViewActionMessageCreator>;
+    let tabStoreData: TabStoreData;
+    let assessmentStoreData: AssessmentStoreData;
+    let reportGeneratorMock: IMock<ReportGenerator>;
+    let visualizationScanResultData: VisualizationScanResultData;
+    let cardsViewData: CardsViewModel;
+    let scanResult: ScanResults;
+
+    beforeEach(() => {
+        featureFlagStoreData = {};
+        actionMessageCreatorMock = Mock.ofType(DetailsViewActionMessageCreator, MockBehavior.Strict);
+        tabStoreData = {
+            title: thePageTitle,
+            url: thePageUrl,
+        } as TabStoreData;
+        assessmentStoreData = {
+            resultDescription: theDescription,
+        } as AssessmentStoreData;
+        assessmentsProviderMock = Mock.ofType<AssessmentsProvider>(undefined, MockBehavior.Loose);
+        reportGeneratorMock = Mock.ofType<ReportGenerator>(undefined, MockBehavior.Strict);
+        cardsViewData = null;
+        scanResult = null;
+    });
+
+    function getProps(): DetailsViewCommandBarProps {
+        const deps = {
+            detailsViewActionMessageCreator: actionMessageCreatorMock.object,
+            getCurrentDate: () => theDate,
+            reportGenerator: reportGeneratorMock.object,
+            getDateFromTimestamp: value => theDate,
+        } as DetailsViewCommandBarDeps;
+
+        visualizationScanResultData = {
+            issues: {
+                scanResult: scanResult,
+            },
+        } as VisualizationScanResultData;
+
+        return {
+            deps,
+            featureFlagStoreData,
+            actionMessageCreator: actionMessageCreatorMock.object,
+            assessmentsProvider: assessmentsProviderMock.object,
+            tabStoreData,
+            assessmentStoreData,
+            visualizationScanResultData,
+            cardsViewData,
+        } as DetailsViewCommandBarProps;
+    }
+
+    function setAssessmentReportGenerator(): void {
+        reportGeneratorMock
+            .setup(x =>
+                x.generateAssessmentReport(
+                    assessmentStoreData,
+                    assessmentsProviderMock.object,
+                    featureFlagStoreData,
+                    tabStoreData,
+                    theDescription,
+                ),
+            )
+            .returns(() => theGeneratorOutput);
+    }
+
+    function setAutomatedChecksReportGenerator(): void {
+        reportGeneratorMock
+            .setup(x =>
```

Comment:
I think we should have a name that identifies that the `x` here is a reportGenerator.

Reply:
Done